daemon/daemon_main.go

```diff
@@ -1101,6 +1091,42 @@ func runDaemon() {
 
 	d.initK8sSubsystem()
 
+	// If K8s is enabled we can do the service translation automagically by
+	// looking at services from k8s and retrieve the service IP from that.
+	// This makes cilium to not depend on kube dns to interact with etcd operator
+	var goopts *kvstore.ExtraOptions
+	if k8s.IsEnabled() {
+		goopts = &kvstore.ExtraOptions{}
+		goopts.DialOption = []grpc.DialOption{
+			grpc.WithDialer(func(s string, duration time.Duration) (conn net.Conn, e error) {
+				// typical service name "cilium-etcd-client.kube-system.svc"
+				fullFQDN := bytes.Split([]byte(s), []byte{'.'})
```

Comment:
Why can't we just do a split operation on `s` as a string? Converting to bytes (and then to strings again) seems to be an additional effort.

Reply:
Yeah, you were right. I've changed this entirely.